### PR TITLE
use gcm uri for notification end points

### PR DESCRIPTION
**Pull request by kashif**
PR for issue #2 

* use gcm uri for notification end point
* added recover_notification_key api

### CODE REVIEW

**Comment by kashif** [comment]
@sabman can you review?

**Comment by sabman** [comment]
Ok I will look tomorrow

File: lib/fcm.rb [hunk 1]
```diff
@@ -120,6 +145,13 @@ def send_to_topic(topic, options = {})
 
   private
 
+  def for_uri(uri)
+    current_uri = self.base_uri
+    self.base_uri uri
```
Comment (sabman) [inline lib/fcm.rb]: Just a quick look should this be `self.base_url = ...`? And one after yield?

Comment (sabman) [inline lib/fcm.rb]: @kashif ignore that just noticed it an `HTTParty` method call.

**Comment by sabman** [comment]
@spacialdb/owners if someone wants to try it on a server that would be ideal. We can merge it after a couple of days. ref #2

**Pull request merged**